PermanentOrg/web-app
# Upgrade to Angular 17

Upgrade Angular to version 17, and upgrade all dependent packages to compatible versions as well.

This upgrade was actually quite small; I mainly ran into issues with testing it and running/upgrading Storybook. Only one file had to be manually changed and only two files were upgraded by automatic Angular migrations.

**This upgrade now requires Node 18 to be installed.**

**Steps to Test:**
1. Run `npm install`
2. Run `npm run build` / `npm run build:local` to verify that a production build can be made without errors.
3. Run unit tests to verify that they can be run without errors and no functionality covered by tests is broken.
4. Run `npm run storybook` to verify Storybook can be run without errors <small>(I always forget this one!)</small>
5. Run `npm run dev` to verify that the local development server can be run without errors
  - Click around in the web-app to make sure nothing's broken!

Resolves #345. Resolves PER-9430.
